drivers/display/display_dummy.c

```diff
@@ -126,5 +126,5 @@ static const struct display_driver_api dummy_display_api = {
 DEVICE_DEFINE(dummy_display, CONFIG_DUMMY_DISPLAY_DEV_NAME,
 		    &dummy_display_init, NULL,
 		    &dummy_display_data, NULL,
-		    APPLICATION, CONFIG_APPLICATION_INIT_PRIORITY,
```

Comment:
we should probably introduce a display init priority together with this change, keeping `CONFIG_APPLICATION_INIT_PRIORITY` doesn't make much sense.

Reply:
I thought about it but decided not to change too many things at a time. I can submit another PR just for the display init priority. Or change add it in this PR if you want.

Reply:
In general I feel that priority changes have no relation with the library changes (maybe I have missed something, though), so they may be part of another PR.

Reply:
let's leave that for a 2nd PR then, I think it's fine.